environment.py

```diff
@@ -0,0 +1,152 @@
+    for file in graph_files:
+        graphs = get_graphs_from_file(file)
+        isomorphs, iso_time, automorphs, auto_time = process_graphs(graphs)
+        result_string = stringify_result(file, isomorphs, iso_time, automorphs, auto_time)
+        print_result(result_string)
+
+
+def get_graphs_from_file(file: str) -> List[Graph]:
+    with open(file) as f:
+        L = load_graph(f, read_list=True)
+    return L[0]
+
+
+def process_graphs(graphs: List[Graph]) -> Tuple[List[List[Graph]], float, List[int], float]:
+    output_result(create_title_string("ISOMORPHISMS"))
+    iso_start = time.time()
+    isomorphs = calculate_isomorphisms(graphs)
+    iso_time = time.time() - iso_start
+
+    output_result(create_title_string("AUTOMORPHISMS"))
+    auto_start = time.time()
+    automorphs = calculate_automorphisms(isomorphs)
+    auto_time = time.time() - auto_start
+    return isomorphs, iso_time, automorphs, auto_time
+
+
+def calculate_isomorphisms(graphs: List[Graph]) -> List[List[Graph]]:
+    isomorphs = [[]]
+    isomorphs[0] = [graphs.pop(0)]
+    for i in range(len(graphs)):
+        added = False
+        start_time = time.time()
+        for j in range(len(isomorphs)):
+            if is_isomorphisms(isomorphs[j][0], graphs[i]):
+                isomorphs[j].append(graphs[i])
+                added = True
+                output_result(graphs[i].name + " and " + isomorphs[j][0].name + " are isomorphisms (" + str(
+                    round(time.time() - start_time, 10)) + "s)")
+                break
+        if not added:
+            isomorphs.append([graphs[i]])
+            output_result(graphs[i].name + " has no isomorphisms yet (" + str(round(time.time() - start_time, 10)) + "s)")
+    return isomorphs
+
+
+def calculate_automorphisms(isomorphs: List[List[Graph]]) -> List[int]:
+    automorphisms = []
+    for isomorph in isomorphs:
+        start_time = time.time()
+        graph = isomorph[0]
+        num_automorphisms = get_number_automorphisms(graph)
+        automorphisms.append(num_automorphisms)
+        output_result(graph.name + "\'s group has " + str(num_automorphisms) + " automorphisms (" + str(
+            round(time.time() - start_time, 10)) + "s)")
+    return automorphisms
+
+
+def stringify_result(file: str, isomorphs: List[List[Graph]], iso_time: float, automorphs: List[int],
+                     auto_time: float) -> str:
+    title_string = create_title_string(get_file_title(file))
+    data_string = create_data_string(isomorphs, automorphs)
+    footer_string = create_footer_string(iso_time, auto_time)
+    return title_string + '\n' + data_string + '\n' + footer_string + '\n'
+
+
+def create_title_string(title: str) -> str:
+    return '\n' + title + '\n' + '{:_<41}'.format('') + '\n'
+
+
+def get_file_title(file: str) -> str:
+    return file.rsplit('/', 1)[1]
+
+
+def create_data_string(isomorphs: List[List[Graph]], automorphs: List[int]) -> str:
+    output_format = '{:<20} {:>20}'
```

Comment:
Wat doen `:<20` en `:>20`?

Reply:
Format voor printen, zoals bij een fprintf of sprintf en dat soort dingen. De breedte van de output ligt dan vast, in dit geval 20 tekens en de ene is rechts uitgelijnd

Reply:
Vandaar dat het er zo strak uit ziet! 😎